TestGoPoolWithMutex 测试用例添加 WithTimeout 无法通过 race 测试

此用例添加 `WithTimeout(10*time.Microsecond)`
``` go
func TestGoPoolWithMutex(t *testing.T) {
	pool := NewGoPool(100, WithLock(new(sync.Mutex)), WithTimeout(10*time.Microsecond))
	defer pool.Release()
	for i := 0; i < 1000; i++ {
		pool.AddTask(func() (interface{}, error) {
			time.Sleep(10 * time.Millisecond)
			return nil, nil
		})
	}
	pool.Wait()
}
```
报错信息：
``` bash
$ go test -v -race ./...
=== RUN   TestGoPoolWithMutex
==================
WARNING: DATA RACE
Write at 0x00c000032670 by goroutine 115:
  github.com/devchat-ai/gopool.(*worker).executeTaskWithTimeout.func1()
      /workspaces/gopool/worker.go:61 +0x6d

Previous write at 0x00c000032670 by goroutine 13:
  github.com/devchat-ai/gopool.(*worker).executeTaskWithTimeout()
      /workspaces/gopool/worker.go:72 +0x344
  github.com/devchat-ai/gopool.(*worker).executeTask()
      /workspaces/gopool/worker.go:39 +0xc4
  github.com/devchat-ai/gopool.(*worker).start.func1()
      /workspaces/gopool/worker.go:26 +0xaa

Goroutine 115 (running) created at:
  github.com/devchat-ai/gopool.(*worker).executeTaskWithTimeout()
      /workspaces/gopool/worker.go:60 +0x275
  github.com/devchat-ai/gopool.(*worker).executeTask()
      /workspaces/gopool/worker.go:39 +0xc4
  github.com/devchat-ai/gopool.(*worker).start.func1()
      /workspaces/gopool/worker.go:26 +0xaa

Goroutine 13 (running) created at:
  github.com/devchat-ai/gopool.(*worker).start()
      /workspaces/gopool/worker.go:23 +0xf7
  github.com/devchat-ai/gopool.NewGoPool()
      /workspaces/gopool/gopool.go:75 +0x54f
  github.com/devchat-ai/gopool.TestGoPoolWithMutex()
      /workspaces/gopool/gopool_test.go:14 +0xd4
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1576 +0x216
  testing.(*T).Run.func1()
      /usr/local/go/src/testing/testing.go:1629 +0x47
==================
==================
WARNING: DATA RACE
Write at 0x00c000032680 by goroutine 115:
  github.com/devchat-ai/gopool.(*worker).executeTaskWithTimeout.func1()
      /workspaces/gopool/worker.go:61 +0xa6

Previous write at 0x00c000032680 by goroutine 13:
  github.com/devchat-ai/gopool.(*worker).executeTaskWithTimeout()
      /workspaces/gopool/worker.go:72 +0x384
  github.com/devchat-ai/gopool.(*worker).executeTask()
      /workspaces/gopool/worker.go:39 +0xc4
  github.com/devchat-ai/gopool.(*worker).start.func1()
      /workspaces/gopool/worker.go:26 +0xaa

Goroutine 115 (running) created at:
  github.com/devchat-ai/gopool.(*worker).executeTaskWithTimeout()
      /workspaces/gopool/worker.go:60 +0x275
  github.com/devchat-ai/gopool.(*worker).executeTask()
      /workspaces/gopool/worker.go:39 +0xc4
  github.com/devchat-ai/gopool.(*worker).start.func1()
      /workspaces/gopool/worker.go:26 +0xaa

Goroutine 13 (running) created at:
  github.com/devchat-ai/gopool.(*worker).start()
      /workspaces/gopool/worker.go:23 +0xf7
  github.com/devchat-ai/gopool.NewGoPool()
      /workspaces/gopool/gopool.go:75 +0x54f
  github.com/devchat-ai/gopool.TestGoPoolWithMutex()
      /workspaces/gopool/gopool_test.go:14 +0xd4
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1576 +0x216
  testing.(*T).Run.func1()
      /usr/local/go/src/testing/testing.go:1629 +0x47
==================
    testing.go:1446: race detected during execution of test
--- FAIL: TestGoPoolWithMutex (0.10s)
=== RUN   TestGoPoolWithSpinLock
--- PASS: TestGoPoolWithSpinLock (0.11s)
=== RUN   TestGoPoolWithError
--- PASS: TestGoPoolWithError (0.11s)
=== RUN   TestGoPoolWithResult
--- PASS: TestGoPoolWithResult (0.10s)
=== RUN   TestGoPoolWithRetry
--- PASS: TestGoPoolWithRetry (0.10s)
=== NAME  
    testing.go:1446: race detected during execution of test
FAIL
FAIL    github.com/devchat-ai/gopool    0.542s
FAIL
```

--- a/gopool.go
+++ b/gopool.go
@@ -84,9 +84,17 @@ func (p *goPool) AddTask(t task) {
 	p.taskQueue <- t
 }
 
-// Wait waits for all tasks to be dispatched.
+// Wait waits for all tasks to be dispatched and completed.
 func (p *goPool) Wait() {
-	for len(p.taskQueue) > 0 {
+	for {
+		p.lock.Lock()
+		workerStackLen := len(p.workerStack)
+		p.lock.Unlock()
+
+		if len(p.taskQueue) == 0 && workerStackLen == len(p.workers) {
+			break
+		}
+
 		time.Sleep(100 * time.Millisecond)
 	}
 }

--- a/gopool_test.go
+++ b/gopool_test.go
@@ -149,3 +149,26 @@ func TestGoPoolWithRetry(t *testing.T) {
 		t.Errorf("Expected task to run %v times, but it ran %v times", retryCount+1, taskRunCount)
 	}
 }
+
+func TestGoPoolWithTimeout(t *testing.T) {
+	var taskRun int32
+
+	pool := NewGoPool(100, WithTimeout(100*time.Millisecond), WithErrorCallback(func(err error) {
+		if err.Error() != "task timed out" {
+			t.Errorf("Expected error 'task timed out', but got %v", err)
+		}
+		atomic.StoreInt32(&taskRun, 1)
+	}))
+	defer pool.Release()
+
+	pool.AddTask(func() (interface{}, error) {
+		time.Sleep(200 * time.Millisecond)
+		return nil, nil
+	})
+
+	pool.Wait()
+
+	if atomic.LoadInt32(&taskRun) == 0 {
+		t.Errorf("Expected task to run and timeout, but it did not run")
+	}
+}

--- a/worker.go
+++ b/worker.go
@@ -54,17 +54,25 @@ func (w *worker) executeTaskWithTimeout(t task, pool *goPool) (result interface{
 	defer cancel()
 
 	// Create a channel to receive the result of the task
-	done := make(chan struct{})
+	resultChan := make(chan interface{})
+	errChan := make(chan error)
 
 	// Run the task in a separate goroutine
 	go func() {
-		result, err = t()
-		close(done)
+		res, err := t()
+		select {
+		case resultChan <- res:
+		case errChan <- err:
+		case <-ctx.Done():
+			// The context was cancelled, stop the task
+			return
+		}
 	}()
 
 	// Wait for the task to finish or for the context to timeout
 	select {
-	case <-done:
+	case result = <-resultChan:
+		err = <-errChan
 		// The task finished successfully
 		return result, err
 	case <-ctx.Done():

